mllib/src/main/scala/org/apache/spark/ml/feature/HashingTF.scala

```diff
@@ -93,11 +97,21 @@ class HashingTF @Since("1.4.0") (@Since("1.4.0") override val uid: String)
   @Since("2.0.0")
   override def transform(dataset: Dataset[_]): DataFrame = {
     val outputSchema = transformSchema(dataset.schema)
-    val hashingTF = new feature.HashingTF($(numFeatures)).setBinary($(binary))
-    // TODO: Make the hashingTF.transform natively in ml framework to avoid extra conversion.
-    val t = udf { terms: Seq[_] => hashingTF.transform(terms).asML }
+    val hashUDF = udf { (terms: Seq[_]) =>
+      val ids = terms.map { term =>
```

Comment:
Why did you implement this differently than the old one?

Reply:
Sorry, I can't remember all details exactly since the pr is too old. if my memory is correct, the ML implementation keep consistent with MLLIB (old one). As the "TODO" above said
> Make the hashingTF.transform natively in ml framework to avoid extra conversion.

Reply:
Ok, I'm just wondering why you changed the code? The old one uses a mutable hashmap, which is different than the approach here  (and is slower, in my tests).

Reply:
Sorry for the delay. Yes, original implement use mutable map in mllib package, which is different with the PR. However, I think the PR is more clear and their logic are the same. Which one is faster in your test?

Reply:
By the way, as the PR doesn't change the function's behavior, I think it should be correct if all tests pass, right?

Reply:
@sethah Hi, thank all for your review and comments. However, since it has been a quite long time with no activity, is it a good idea to close the PR?